### PR TITLE
Add `Control.Foldl.NonEmpty.toFold`

### PR DESCRIPTION
**Pull request by Gabriella439**
Fixes https://github.com/Gabriella439/foldl/issues/190

### CODE REVIEW

**Comment by Profpatsch** [comment]
Wow, this is tricky! No wonder I couldn’t write it in like 10 minutes yesterday.

**Comment by Profpatsch** [comment]
Shouldn’t the Either be a strict Either? Or is that not relevant for the step function?

**Comment by Gabriella439** [comment]
Yeah, at least the `Right` branch at least should probably be strict

**Review by Profpatsch — APPROVED** [review]

File: src/Control/Foldl/NonEmpty.hs [hunk 1]
```diff
@@ -155,6 +156,20 @@ fromFold :: Fold a b -> Fold1 a b
 fromFold (Fold step begin done) = Fold1 (\a -> Fold step (step begin a) done)
 {-# INLINABLE fromFold #-}
 
+-- | Promot any `Fold1` to an equivalent `Fold`
```
Comment (Profpatsch) [inline src/Control/Foldl/NonEmpty.hs:159]: Promote (typo)

**Pull request merged**